src/clj/fluree/db/json_ld/policy/rules.cljc

```diff
@@ -84,68 +72,106 @@
      policy-map
      (:on-property restriction-map))))
 
-(defn restriction-map
-  [restriction]
-  (let [id          (util/get-id restriction) ;; @id name of restriction
-        on-property (when-let [props (util/get-all-ids restriction const/iri-onProperty)]
-                      (set props)) ;; can be multiple properties
-        on-class    (when-let [classes (util/get-all-ids restriction const/iri-onClass)]
-                      (set classes))
-        src-query   (util/get-first-value restriction const/iri-query)
-        query       (if (map? src-query)
+(defn parse-targets
+  [db target-exprs]
+  (let [out-ch (async/chan 2)
+        in-ch  (async/to-chan! (mapv #(or (util/get-id %) (util/get-value %)) target-exprs))]
+    (async/pipeline-async 2
+                          out-ch
+                          (fn [target-expr ch]
+                            (if (map? target-expr)
+                              ;; maps are query where clauses
+                              (-> (dbproto/-query db (assoc target-expr "select" "?$target"))
+                                  (async/pipe (async/chan 2 cat))
+                                  (async/pipe ch))
+                              ;; non-maps are literals
+                              (async/onto-chan! ch [target-expr])))
+                          in-ch)
+    (async/into #{} out-ch)))
+
+(defn parse-policy
+  [db policy-doc]
+  (go-try
+    (let [id (util/get-id policy-doc) ;; @id name of policy-doc
+
+          subject-targets  (when-let [target-exprs (get policy-doc const/iri-targetSubject)]
+                             (<? (parse-targets db target-exprs)))
```

Comment:
Per prior comments, I think we should encode the return IRIs when parsing... perhaps here or down below in the fn.

What if @context is used in the query? If so, you won't be able to encode the IRIs for comparison as they will already be "shortened".

For this reason, I added a `:selection-context` internal field in a different PR which, if present, gets used in the select statement. It can be used to remove @context from applying to `select` statements.